projects/graph/graph.py

```diff
@@ -11,37 +17,150 @@ def add_vertex(self, vertex):
         """
         Add a vertex to the graph.
         """
+        self.vertices[vertex] = set()
         pass  # TODO
     def add_edge(self, v1, v2):
         """
         Add a directed edge to the graph.
         """
+        if v1 not in self.vertices:
```

Comment:
I wouldn't have the add edge method creating verticies if they do not exist. This should be strictly just creating an edge between them. So I would remove the add part and raise an error if they do not exist